README.md

```diff
@@ -273,75 +261,128 @@ By default, these files are prefixed by the name of the Spanner database (with a
-drivers are _'postgres'_, _'pg_dump'_, _'mysql'_ and _'mysqldump'_. By default, 
-the driver is _'pg_dump'_.
+#### harbourbridge `data`
 
-`-schema-sample-size` Specifies the number of rows to use for inferring schema 
-(only for DynamoDB). By default, the schema sample size is 100,000.
+This subcommand will perform data migration and report on the quality of the same. Rows which could not be migrated are reported in
+dropped.txt file. This subcommand requires users to pass the session file (which contains schema mapping) generated by either the `schema` subcommand or web UI.
+
+#### harbourbridge `oneshot`
+
+This subcommand will generate a schema as well as perform data migration and report on the quality of both schema migration and data migration. This subcommand can be used to do a quick evaluation for the migration and get started quickly on Spanner.
+
+### Command line flags
+
+This section describes the flags common across all the subcommands. For flags
+specific to a give subcommand run `harbourbridge help <subcommand>`.
+
+`-source` Required flag. Specifies the source database. Supported source
+databases are _'postgres'_, _'mysql'_ and _'dynamodb'_.
+
+`-target` Optional flag. Specifies the target database. Defaults to _'spanner'_
+, which is the only supported target database today.
 
 `-prefix` Specifies a file prefix for the report, schema, and bad-data files
 written by the tool. If no file prefix is specified, the name of the Spanner
 database (plus a '.') is used.
 
-`-v` or `-verbose` Specifies verbose mode. This will cause HarbourBridge to output detailed
-messages about the conversion.
-
-`-schema-only` Specifies that only schema processing will be performed.
-Any data in the source database will be ignored.
-
-`-data-only` Specifies that only data migration will be performed.
-A spanner database will be created based on the schema state provided
-by a session file (`-session`) and data will be migrated.
+`-v` or `-verbose` Specifies verbose mode. This will cause HarbourBridge to
+output detailed messages about the conversion.
 
 `-skip-foreign-keys` Controls whether we add foreign key constraints after
 data migration is complete. This flag cannot be used with schema-only mode,
 and does not affect the generation of foreign key statements during schema
-processing i.e. foreign key constraints will still appear in the generated 
+processing i.e. foreign key constraints will still appear in the generated
 Spanner DDL files.
 
-`-session` Specifies a session file that contains all schema and data 
+`-session` Specifies a session file that contains all schema and data
 conversion state endcoded as JSON.
 
-`-dump-file` Specifies the path where the dump file is stored. 
-If the location is Google Cloud Storage (GCS), you can use the following format:
-`-dump-file=gs://{bucket_name}/{path/to/dump_file}`. Please ensure you have
-read pemissions to the GCS bucket you would like to use. 
+#### `-source-profile`
```

Comment:
The formatting of this looks kind of odd: there is a list of command line flags, and then it continues as a header.
Suggest the following change: extend the Command line flags section to be a full list of flags i.e add:

`-source-profile` Specifies detailed parameters for the source database such as connection parameters. See Source Profile (linked to section) for details.

`-target-profile` Specifies detailed parameters for the target database. See Target Profile (linked to section) for details.

And then rename the sections "Source Profile" and "Target Profile" respectively.

Reply:
Hmm, I did spend quite some thinking about how to structure this, but couldn't make it clean. Makes sense to link to sections, thanks for the advice. Done.